Library/Homebrew/livecheck/strategy/xorg.rb

```diff
@@ -84,15 +82,12 @@ def self.find_versions(url, regex)
 
           regex ||= /href=.*?#{Regexp.escape(module_name)}[._-]v?(\d+(?:\.\d+)+)\.t/i
 
-          match_data = { matches: {}, regex: regex, url: page_url }
-
-          # Cache responses to avoid unnecessary duplicate fetches
-          @page_data[page_url] = URI.parse(page_url).open.read unless @page_data.key?(page_url)
+          # Use the cached page content to avoid duplicate fetches
+          cached_content = @page_data[page_url]
+          match_data = PageMatch.find_versions(page_url, regex, cached_content)
 
-          matches = @page_data[page_url].scan(regex)
-          matches.map(&:first).uniq.each do |match|
-            match_data[:matches][match] = Version.new(match)
-          end
+          # Cache any new page content
+          @page_data[page_url] = match_data[:content] if match_data[:content].present?
```

Comment:
Hm... it looks like `page_data` is used before it's defined/created ? 😕

Reply:
Is the `@page_data = {}` definition below `URL_MATCH_REGEX` not sufficient?

Reply:
It is! :) missed it... thanks